### PR TITLE
state serviceaccount setting

### LINKED ISSUE #158: The loki helm chart produces conflicting yaml when creating a serviceAccount with a specified name
When using v0.0.1 of the loki helm chart, if you specify:
```
serviceAccount:
  create: true
  name: not-empty-string
```
then two serviceAccounts are created, both with the name 'not-empty-string'. This doesn't work.

This could be resolved by allowing the user to specify whether to create the serviceAccount + the name to use/create separately for `loki` and `promtail`.

### PR DESCRIPTION
**Pull request by daixiang0**
Fix #158 

Signed-off-by: Xiang Dai <764524258@qq.com>

### CODE REVIEW

**Comment by laverya** [comment]
I suppose that's one way to fix it 😆 

Should we add that as a comment near https://github.com/grafana/loki/blob/ea712f38b82e6747892475272ff2422b46b0ed15/production/helm/values.yaml#L5 too?

**Comment by jgsqware** [comment]
Maybe, we can not let thing failing and just create the service account in loki template, as part of this helm chart both are created in the same time.

**Comment by daixiang0** [comment]
The logic is create same sa for both loki and promtail.
Do you mean remove sa create part in promtail?

File: production/helm/README.md [hunk 1]
```diff
@@ -17,6 +17,17 @@ $ cd loki/production/helm
...
 
+In `values.yaml`, do not add any string to `name` section if do not want to use custom serviceaccount
+since it would lead to install failed:
+
+```
+serviceAccount:
+  create: true
+  name:
+```
+If not, you must change `craete` section to "false", then set "name" section as you want,
```
Comment (ProPheT777) [inline production/helm/README.md]: s/craete/create

Comment (daixiang0) [reply production/helm/README.md]: thanks!

**Comment by jgsqware** [comment]
Yes, we can only create the sa in Loki itself

**Comment by daixiang0** [comment]
remove promtail sa part and use loki sa.

**Comment by jgsqware** [comment]
LGTM

**Comment by daixiang0** [comment]
any input?

**Comment by daixiang0** [comment]
any update?

**Comment by daixiang0** [comment]
@tomwilkie please review

**Comment by tomwilkie** [comment]
Sorry for the delay!  If @jgsqware is happy, I'm happy - he know more about helm than I do.

**Pull request merged**